File: src/main/java/net/dontdrinkandroot/fixtures/referencerepository/MapBackedReferenceRepository.java
package net.dontdrinkandroot.fixtures.referencerepository;

import java.util.HashMap;
import java.util.Map;

/**
 * {@link ReferenceRepository} that uses a {@link HashMap} as storage.
 *
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class MapBackedReferenceRepository implements ReferenceRepository
{
    private final Map<String, Object> objects = new HashMap<>();

    @Override
    public <T> void store(String name, T object)
    {
        this.objects.put(name, object);
    }

    @Override
    public <T> T retrieve(String name)
    {
        @SuppressWarnings("unchecked")
        T reference = (T) this.objects.get(name);
        if (null == reference) {
            throw new RuntimeException(String.format("No reference found with name '%s'", name));
        }

        return reference;
    }
}

File: src/main/kotlin/net/dontdrinkandroot/fixtures/referencerepository/MapBackedReferenceRepository.kt
package net.dontdrinkandroot.fixtures.referencerepository

import java.util.*

/**
 * [ReferenceRepository] that uses a [HashMap] as storage.
 */
class MapBackedReferenceRepository : ReferenceRepository {

    private val objects: MutableMap<String, Any> = HashMap()

    override fun <T : Any> store(name: String, obj: T) {
        objects[name] = obj
    }

    override fun <T : Any> retrieve(name: String): T =
        (objects[name] ?: throw RuntimeException("Reference not found")) as T
}